src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -408,7 +410,9 @@ trait Types
      *  for a reference denoting an abstract type, its bounds,
      *  for all other types, a TypeBounds type all of whose bounds are this type.
      */
-    def bounds: TypeBounds = TypeBounds(this, this)
+    def bounds: TypeBounds = TypeBounds(lowerBound, upperBound)
+    def lowerBound: Type = this
```

Comment:
Would introducing these methods in the class `Type` create a compatiblity issue?

Reply:
Only if people are inadvertently using scala-compiler.jar 2.12.9 with scala-reflect.jar 2.12.8. However this isn't a supported scenario.